haystack/super_components/indexers/sentence_transformers_document_indexer.py

```diff
@@ -0,0 +1,195 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Literal, Optional
+
+from haystack import Pipeline, component, default_from_dict, default_to_dict
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder
+from haystack.components.writers import DocumentWriter
+from haystack.core.super_component import SuperComponent
+from haystack.document_stores.types import DocumentStore, DuplicatePolicy
+from haystack.utils import (
+    ComponentDevice,
+    Secret,
+    deserialize_document_store_in_init_params_inplace,
+    deserialize_secrets_inplace,
+)
+from haystack.utils.hf import deserialize_hf_model_kwargs, serialize_hf_model_kwargs
+
+
+@component
+class SentenceTransformersDocumentIndexer(SuperComponent):
+    """
+    A document indexer that takes a list of documents, embeds them using SentenceTransformers, and stores them.
```

Comment:
As I mentioned briefly before, I'm not sure this pre-built component is that useful. It only combines two components together (Sentence Transformer Document Embedder + DocumentWriter) and it is directly tied to Sentence Transformers which I'm not sure is our most popular document embedder. 

I'd consider leaving this one out for now since I think pre-built SuperComponent's shine when many components are logically grouped together like in MultiFileConverter.